src/libs/Navigation/AppNavigator/AuthScreens.js

```diff
@@ -109,11 +118,20 @@ class AuthScreens extends React.Component {
         }, ['meta'], true);
     }
 
-    shouldComponentUpdate(prevProps) {
-        if (prevProps.isSmallScreenWidth !== this.props.isSmallScreenWidth) {
+    shouldComponentUpdate(nextProps) {
+        if (nextProps.isSmallScreenWidth !== this.props.isSmallScreenWidth) {
             return true;
         }
 
+        // Update initialHomeParams only once
+        if (!this.initialHomeParams) {
+            // Either we have a reportID or fetchAllReports resolved with no reports. Otherwise keep waiting
+            if (nextProps.initialReportID || nextProps.initialReportID === '') {
```

Comment:
If someone e.g. decided to give a `defaultProps` of empty string to `initialReportID` this would break right?

I don't have a better suggestion yet, but I'm curious if we can make more obvious what is happening here as `initialReportID === ''` feels like an implementation detail for `fetchAllReports resolved with no reports`.

Reply:
Yeah, I thought about this too, but decided to get some feedback from you and review my general direction on this PR

We could hook to the promise here https://github.com/Expensify/Expensify.cash/blob/0874880f20c22f4453a4759e6885e7f28f170ca2/src/libs/Navigation/AppNavigator/AuthScreens.js#L106

And do something like

```js
componentDidMount() {
  fetchAllReports(true, true).finally(() => this.resolveInitialState);
}

resolveInitialState() {
  // If initial nav params haven't resolved up until now just start with sensible defaults
  const routeID = this.props.initialReportID || '';

  // we can return early if initial state is already set
  
  // We'll need to use `setState` to ensure a re-render
  this.setState({
     initialHomeParams: {
        screen: 'Report',
        params: {reportID},
     }
  });
}
```

This above is definitely more explicit than interpreting `nextProps.initialReportID === ''` and comments

shouldComponentUpdate - should account for state updates using the 2nd parameter

Reply:
Hmm yeah this is definitely more explicit and probably how I'd approach this as well, but we also await the promise now (which we have set limits on). I'm unsure if we'd want to break the streak for this.

Another option would be to create something like a `isLoadingInitialReports` flag in Onyx,  but this seems kind of backwards so maybe we can just live with this for now.

Reply:
Sorry but can you please say what do you mean by "this" so I can address it tomorrow
- use the "promise" version 
- leave it the way it is for the time being?

Reply:
No worries, I am saying to leave it like we have it here. We can always add more comments to fill out additional context that might not be obvious.